pkg/provisioner/ironic/clients/client_test.go

```diff
@@ -0,0 +1,230 @@
+			ClientCertificateFile: "",
+			ClientPrivateKeyFile:  "",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	noAuthClient, _ := IronicClient(noAuthClientArgs.ironicEndpoint, noAuthClientArgs.auth, noAuthClientArgs.tls)
+
+	noAuthTlsClientArgs := args{
+		ironicEndpoint: "https://localhost",
+		auth: AuthConfig{
+			Type:     NoAuth,
+			Username: "",
+			Password: "",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "/path/to/ca/file.crt",
+			ClientCertificateFile: "/path/to/cert/file.crt",
+			ClientPrivateKeyFile:  "/path/to/cert/file.key",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	noAuthTlsClient, _ := IronicClient(noAuthTlsClientArgs.ironicEndpoint, noAuthTlsClientArgs.auth, noAuthTlsClientArgs.tls)
+
+	basicAuthClientArgs := args{
+		ironicEndpoint: "http://localhost",
+		auth: AuthConfig{
+			Type:     HTTPBasicAuth,
+			Username: "username",
+			Password: "password",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "",
+			ClientCertificateFile: "",
+			ClientPrivateKeyFile:  "",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	basicAuthClient, _ := IronicClient(basicAuthClientArgs.ironicEndpoint, basicAuthClientArgs.auth, basicAuthClientArgs.tls)
+
+	basicAuthTlsClientArgs := args{
+		ironicEndpoint: "https://localhost",
+		auth: AuthConfig{
+			Type:     HTTPBasicAuth,
+			Username: "username",
+			Password: "password",
+		},
+		tls: TLSConfig{
+			TrustedCAFile:         "/path/to/ca/file.crt",
+			ClientCertificateFile: "/path/to/cert/file.crt",
+			ClientPrivateKeyFile:  "/path/to/cert/file.key",
+			InsecureSkipVerify:    true,
+			SkipClientSANVerify:   true,
+		},
+	}
+	basicAuthTlsClient, _ := IronicClient(basicAuthTlsClientArgs.ironicEndpoint, basicAuthTlsClientArgs.auth, basicAuthTlsClientArgs.tls)
+
+	tests := []struct {
+		name       string
+		args       args
+		wantClient *gophercloud.ServiceClient
+		wantErr    bool
+	}{
+		{
+			name:       "non supported auth type return error",
+			args:       badClientArgs,
+			wantClient: nil,
+			wantErr:    true,
+		},
+		{
+			name:       "noauth auth type without tls return no error",
+			args:       noAuthClientArgs,
+			wantClient: noAuthClient,
+			wantErr:    false,
+		},
+		{
+			name:       "noauth auth type with tls return no error",
+			args:       noAuthTlsClientArgs,
+			wantClient: noAuthTlsClient,
+			wantErr:    false,
+		},
+		{
+			name:       "basicauth auth type without tls return no error",
+			args:       basicAuthClientArgs,
+			wantClient: basicAuthClient,
+			wantErr:    false,
+		},
+		{
+			name:       "basicauth auth type with tls return no error",
+			args:       basicAuthTlsClientArgs,
+			wantClient: basicAuthTlsClient,
+			wantErr:    false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotClient, err := IronicClient(tt.args.ironicEndpoint, tt.args.auth, tt.args.tls)
+			if (err != nil) != tt.wantErr {
```

Comment:
Could you use assertions from the package instead of hand-written checks?

Reply:
Updated